### DISCUSSION TITLE
revertReason : 7AAFAE97 from ERC-4337: Entry Point 0.6.0 when doing a contract call

### DISCUSSION THREAD

**Discussion by EonJeti** [reown-com/appkit#2665] [Q&A]
I have begun using the WalletConnect AppKIt to connect our users to our dapp.  Through the testing of it, I have found that connecting through the standard wallets (MetaMask, Phantom, etc) will work as we intend it to, but when using the email and social logins, we get an error when interacting with a contract.

The connection to the dapp works fine, the authentication and establishing that the dapp is trusted.  However, when calling the smart contract, it completes without calling the contract and doesn't throw an error.  I have gone through the logs and have found that I get an error on the ERC-4337 Entry Point contract.

I had been getting another error, but found that I needed to establish the gasLimit in order for it to work so I added that.  Setting that allowed the contract execution to process, but it never got to the contract call itself.

![image](https://github.com/user-attachments/assets/b3493271-e221-4ea3-b993-af475f01cc59)

The above image is an example of what I get.

I have gone through all of the examples and I have implemented everything as I should.

Structure:

- Nextjs
- Node 20
- Ethers V6

Transaction example:

https://polygonscan.com/tx/0xb61d03006be5a36466f9e8c7910fb9e7ca55d72dd3f806b537915727f60cf8fd

Any assistance in this matter would be greatly appreciated.

**Comment by rtomas** [comment]
Hi @EonJeti,
Thanks for reporting it. Do you have a minimal example to test it ?
Thanks

**Reply by EonJeti** [reply]
I'll throw something together.  And just to clarify, all of the read requests work great, the writes to the contracts don't.
